File: Java/JCT/greedy/Change.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Change {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        System.out.print("금액을 입력하세요: ");
        int N = Integer.parseInt(br.readLine());
        System.out.println(getMinCoins(N));
        br.close();
    }

    private static int getMinCoins(int N) {
        int[] coins = { 500, 100, 50, 10 };
        int numOfCoins = 0;

        for (int coin : coins) {
            numOfCoins += N / coin;
            N %= coin;
        }

        return numOfCoins;
    }
}

File: TypeScript/src/JCT/greedy/change.ts
function change() {
    const readline = require('readline');
    const { stdin, stdout } = process;

    let input: number = 0;
    const rl = readline.createInterface({
        input: stdin,
        output: stdout,
    });
    rl.on('line', (line) => {
        input = parseInt(line);
    }).on('close', solution);

    function solution() {
        let N: number = input;
        console.log(getMinCoins(N));
    }

    const getMinCoins = (N: number) => {
        const coins: number[] = [500, 100, 50, 10];
        return coins.reduce((count: number, coin: number) => {
            const addition: number = Math.floor(N / coin);
            N %= coin;
            return count + addition;
        }, 0);
    };
}

change();
